src/mbgl/util/string_indexer.cpp

```diff
@@ -51,20 +66,21 @@ StringIdentity StringIndexer::get(std::string_view string) {
     }
 }
 
-const std::string& StringIndexer::get(const StringIdentity id) {
+std::string_view StringIndexer::get(const StringIdentity id) {
     std::shared_lock<std::shared_mutex> readerLock(instance().sharedMutex);
 
     const auto& identityToString = instance().identityToString;
     assert(id < identityToString.size());
 
-    return id < identityToString.size() ? identityToString[id] : empty;
+    return id < identityToString.size() ? (instance().buffer.data() + identityToString[id]) : empty;
```

Comment:
Have you considered what might happen if the caller holds on to the returned `string_view` while another thread (or even the same one later on) triggers a reallocation of `buffer`?

It might be prudent to couple the read lock with the returned string_view to at least reduce such a possibility.

Reply:
I see the issue, but I don't quite follow the solution. It would be prudent to return a copy?

Reply:
That's a good point, and is another argument for something else we talking about, just using fixed-sized, pre-allocated blocks that never change, at a small cost of unused memory.

There aren't many uses for getting strings by ID outside of debugging, though, so we could just return `std::string` with minimal performance effect.

Reply:
> using fixed-sized, pre-allocated blocks that never change, at a small cost of unused memory.

yeah, turning `buffer` into a vector of buffers would be a cool solution. if the vector of buffers uses move semantics to grow, that is :)

Reply:
Yes, a copy would be the safest option if you feel that is acceptable. I would consider the frequency at which strings are fetched from the buffer to determine if constructing `std::string`s is an acceptable performance trade-off.